src/module/item/physical/document.ts

```diff
@@ -575,6 +577,33 @@ abstract class PhysicalItemPF2e<TParent extends ActorPF2e | null = ActorPF2e | n
             equipped["-=inSlot"] = null;
         }
 
+        // Special handling for ephemeral hit point changes: if the max HP would be adjusted when changing this item's
+        // carry type, proportionally adjust the current HP as well.
+        type MaybeItemAlteration = { key: string; property?: unknown; itemType?: unknown };
+        const hasHpChangeRules =
+            this.actor?.rules.some(
+                (r: MaybeItemAlteration) =>
+                    r.key === "ItemAlteration" && r.property === "hp-max" && r.itemType === this.type
+            ) ?? false;
+        if (hasHpChangeRules && changed.system?.equipped?.carryType) {
+            // Simulate the update to determine whether max hit points will change
+            const postUpdateHPMax = ((): number => {
+                const actorSource = this.actor!.toObject();
```

Comment:
I think we can avoid the ! if actor is assigned to a variable first. Actor returning form a getter seems to be the reason TS doesn't understand that hasHpChangeRules also validates that there's an actor.

Reply:
![image](https://github.com/foundryvtt/pf2e/assets/106829671/22b3b11a-adc4-47e2-bd8f-32b226028af6)

Reply:
silly TS